DiaHelperApplication/DiaHelperApplication/Controllers/HomeViewController.swift

```diff
@@ -67,61 +103,185 @@ class HomeViewController: UIViewController{
+        
+        meals.append(Meal(timestamp: Date(), bloodSugar: 5.7, insulinDose: 8, carbsIntake: 47.5, foodType: .slow))
         
-        addSubviews()
-        addViewConstraints()
         
     }
     
+    private func setupTableView() {
+        tableView.dataSource = self
+        tableView.delegate = self
+        tableView.register(HomeVcMealTableViewCell.self, forCellReuseIdentifier: "Cell")
+        tableView.translatesAutoresizingMaskIntoConstraints = false
+        
+        self.tableView.layer.borderColor = UIColor.systemGray6.cgColor
+        self.tableView.layer.borderWidth = 1;
+        self.tableView.layer.cornerRadius = 10;
+        
+        tableViewBottomConstraint = tableView.bottomAnchor.constraint(equalTo: tableView.topAnchor, constant: CGFloat((44 * (meals.count + 1))))
+        tableViewBottomConstraint.isActive = true
+    }
+    
+    
     private func setupAddButton() {
         let addAction = UIAction(handler: addButtonTapped)
         addButton.addAction(addAction, for: .touchUpInside)
     }
     
-    
     private func addButtonTapped(_ action: UIAction) {
         let addNutritionViewController = AddNutritionViewController()
+        addNutritionViewController.delegate = self
         let navController = UINavigationController(rootViewController: addNutritionViewController)
         navController.modalPresentationStyle = .fullScreen
         navigationController?.present(navController, animated: true)
     }
     
+    private func glucometerAction() {
+        glucometerButton.addTarget(self, action: #selector(self.glucometterButtonTapped), for: .touchUpInside)
+    }
+    
+    @objc func glucometterButtonTapped(sender: UIButton){
+        let glucometerValueViewController = GlucometerValueViewController()
+        glucometerValueViewController.delegate = self
+        let navController = UINavigationController(rootViewController: glucometerValueViewController)
+        navController.modalPresentationStyle = .fullScreen
+        navigationController?.present(navController, animated: true)
+    }
+    
     private func addSubviews() {
-        view.addSubview(bloodSugar)
+        bloodSugarContainer.addSubview(bloodSugar)
+        view.addSubview(bloodSugarContainer)
         view.addSubview(addButton)
-
+        view.addSubview(mealsLabel)
+        view.addSubview(glucometerButton)
+        view.addSubview(tableView)
     }
     
     func addViewConstraints() {
         NSLayoutConstraint.activate([
-            bloodSugar.leadingAnchor.constraint(equalTo: view.safeAreaLayoutGuide.leadingAnchor, constant: 16),
-            bloodSugar.trailingAnchor.constraint(equalTo: view.safeAreaLayoutGuide.trailingAnchor, constant: -16),
-            bloodSugar.topAnchor.constraint(equalTo: view.safeAreaLayoutGuide.topAnchor, constant: 16)
-        ])
-        
-        NSLayoutConstraint.activate([
+            bloodSugarContainer.widthAnchor.constraint(equalToConstant: 64),
```

Comment:
The blood sugar items are not centred on the different screen sizes.
One way is to constraint the `bloodSugarContainer.trailingAnchor` to the safe area and center the `mealsLabel` and fix the `glucometerButton` to the trailing or better to replace the container with a StackView which will manage everything.

<img width="966" alt="Screenshot 2023-12-28 at 17 46 40" src="https://github.com/petyadamyanova/DiaHelperApp/assets/1854038/82389925-52da-458d-9d81-9c8d7d35fee2">